samples/star-wars-api/Program.fs

```diff
@@ -2,13 +2,39 @@ namespace FSharp.Data.GraphQL.Samples.StarWarsApi
 
 open Microsoft.AspNetCore
 open Microsoft.AspNetCore.Hosting
+open Microsoft.Extensions.Configuration
+
 
 module Program =
     let exitCode = 0
 
-    let buildWebHost args =
+    let buildWebHost (args: string array) =
+
+        // Build an initial configuration that takes in both environment and command line settings.
+        let config = ConfigurationBuilder()
+                            .AddEnvironmentVariables()
+                            .AddCommandLine(args)
+                            .Build()
+
+        // This is done so that an environment specified on the command line with "--environment" is respected,
+        // when we look for appsettings.*.json files.
+        let configureAppConfiguration (context: WebHostBuilderContext) (config: IConfigurationBuilder) =
+
+            // The default IFileProvider has the working directory set to the same as the DLL.
+            // We'll use this to re-set the FileProvider in the configuration builder below.
+            let fileProvider = ConfigurationBuilder().GetFileProvider()
+
+            // Extract the environment name from the configuration that was already built above
+            let envName = context.HostingEnvironment.EnvironmentName
+            // Use the name to find the additional appsettings file, if present.
+            config.SetFileProvider(fileProvider)
+                    .AddJsonFile("appsettings.json", false, true)
+                    .AddJsonFile($"appsettings.{envName}.json", true) |> ignore
```

Comment:
Then we do not need this

Reply:
https://github.com/dotnet/aspnetcore/blob/6796d5b2250cfd279def809f70b971a3b2b791da/src/DefaultBuilder/src/WebHost.cs#L154

Reply:
Same issue as before...  It's not using the "development" appsettings when the "development" environment is specified on the command line.

Note there's no trace output in the app:

```
dotnet tests/FSharp.Data.GraphQL.IntegrationTests.Server/bin/Debug/net6.0/FSharp.Data.GraphQL.IntegrationTests.Server.dll --environment Development --configuration Debug --urls=http://localhost:8085/ (In: true, Out: false, Err: false)
Hosting environment: Development
Content root path: /Users/garrett/Documents/risk-assessment/FSharp.Data.GraphQL
Now listening on: http://localhost:8085
Application started. Press Ctrl+C to shut down.
Finished (Success) 'StartIntegrationServer' in 00:00:03.2215225
Starting target 'UpdateIntrospectionFile'
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 GET http://localhost:8086/ - -
info: FSharp.Data.GraphQL.Samples.StarWarsApi.HttpHandlers[0]
      Produced direct GraphQL response with documentId = '1731438334' and metadata:
      map [("filters", []); ("queryWeight", 0.0); ("queryWeightThreshold", 2.0)]
info: Microsoft.AspNetCore.Http.Result.OkObjectResult[1]
      Writing value of type 'FSharp.Data.GraphQL.GQLResponse' with status code '200'.
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished HTTP/1.1 GET http://localhost:8086/ - - - 200 - application/json;+charset=utf-8 416.0309ms```

I'd like to keep this code in for now, because I often use a command line flag when I'm debugging locally.  I should probably also file a ticket against `defaultBuilder`, because this really should work...